src/PFire.Core/Session/XFireClient.cs

```diff
@@ -25,16 +31,24 @@ public class XFireClient
 
         public XFireClient(TcpClient tcpClient)
         {
+            
             TcpClient = tcpClient;
             
             // TODO: be able to use unique salts
             Salt = "4dc383ea21bf4bca83ea5040cb10da62";//Guid.NewGuid().ToString().Replace("-", string.Empty);
             SessionId = Guid.NewGuid();
+            Initialized = true;
```

Comment:
This is incorrect behaviour.. the client is only initialized once the connected client sends the 4 byte header.